src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -675,11 +682,11 @@ private static string Encode5987(StringSegment input)
                                    '0', '1', '2', '3', '4', '5', '6', '7',
                                    '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };
 
-    private static void HexEscape(StringBuilder builder, char c)
+    private static void HexEscape(StringBuilder builder, byte b)
     {
         builder.Append('%');
-        builder.Append(HexUpperChars[(c & 0xf0) >> 4]);
-        builder.Append(HexUpperChars[c & 0xf]);
+        builder.Append(HexUpperChars[(b & 0xf0) >> 4]);
+        builder.Append(HexUpperChars[b & 0xf]);
```

Comment:
`builder.Append($"%{HexUpperChars[(b & 0xf0) >> 4]}{HexUpperChars[b & 0xf]}")` is probably in theory faster.

Reply:
But it allocates more, no?

Reply:
No, it uses the string interpolator overload

Reply:
I implemented posted the results. I could not measure a difference with the perf test.